src/icalendar/cal.py

```diff
@@ -436,6 +436,22 @@ def __repr__(self):
         subs = ', '.join(str(it) for it in self.subcomponents)
         return f"{self.name or type(self).__name__}({dict(self)}{', ' + subs if subs else ''})"
 
+    def __eq__(self, other):
+        properties_equal = super().__eq__(other)
+        if not properties_equal:
+            return False
+
+        # The subcomponents might not be in the same order,
+        # neither there's a natural key we can sort the subcomponents by nor
+        # are the subcomponent types hashable, so  we cant put them in a set to
+        # check for set equivalence. We have to iterate over the subcomponents
+        # and look for each of them in the list.
+        for subcomponent in self.subcomponents:
+            if subcomponent not in other.subcomponents:
```

Comment:
Copy the calendar and remove an event and test again a==b and b==a.

Reply:
`all(component in other.subcomponents for component in self.subcomponents) and all(component in self.subcomponents for component in other.subcomponents)` but this has O(2^n) which is bad.

ah!
return len(self.subcomponents) == len(other.subcomponents) 
should do the same.

Reply:
Good catch!